### PR TITLE
Added MariaDB support

### PR DESCRIPTION
**Pull request by Maescool**
Added MariaDB support based on sqlite implementation.

### CODE REVIEW

File: common/src/main/resources/config.yml [hunk 1]
```diff
@@ -38,13 +38,20 @@ player-link:
...
+    # The database type you want to use.
+    # The option is only used when auto-download is enabled or when
+    # there are more then one database implementations found in the config directory
+    # current options: sqlite, mariadb
+    type: sqlite
+    # These settings are not used for sqlite, but are for others (like mariadb)
+    database: floodgate
+    username: floodgate
+    password: P4ssw0rd
+    hostname: localhost
```
Comment (rtm516) [inline common/src/main/resources/config.yml:51]: As below. (why these are in the wrong order I don't know)

File: common/src/main/java/org/geysermc/floodgate/config/FloodgateConfig.java [hunk 2]
```diff
@@ -70,7 +70,16 @@ public boolean isProxy() {
...
+        private PlayerLinkDatabase database;
+    }
+
+    @Getter
+    public static class PlayerLinkDatabase {
         private String type;
+        private String database;
+        private String username;
+        private String password;
+        private String hostname;
```
Comment (rtm516) [inline common/src/main/java/org/geysermc/floodgate/config/FloodgateConfig.java:82]: I feel like these should somehow be dynamically be included, maybe a separate config file just for the db as not all connections will have the same options.

Comment (Maescool) [reply common/src/main/java/org/geysermc/floodgate/config/FloodgateConfig.java:82]: I was thinking the same.. but wasn't 100% sure how to handle it.. this was the fastest way to make it work cleanly

File: common/src/main/resources/config.yml (hunk 1 above)
Comment (Maescool) [reply common/src/main/resources/config.yml:51]: idk where my mind was.. but it doesn't break anything xD

**Pull request closed without merging**